### PR TITLE
Add Cassandra Vector Memory provider + gated integration tests

### PR DESCRIPTION
**Pull request by aykhanpashayev**
# Description

This PR adds a new **Cassandra-backed Vector Memory provider** to `bx-ai`, along with configuration/unit tests, an opt-in end-to-end integration test, and developer tooling/docs to make local validation straightforward.

Primary goals:
- Provide a Cassandra 5.0+/Astra-compatible Vector Memory implementation.
- Keep default CI/local `./gradlew test` green by **gating external-service tests** behind explicit environment toggles.
- Improve dev ergonomics (devcontainer + schema docs) and fix Gradle/JAR resolution edge cases for development builds.

---

### Cassandra Vector Memory
- Added `CassandraVectorMemory` implementation under `src/main/bx/models/memory/vector/`.
- Updated `aiMemory.bx` to support `memory: "cassandra"` / `CassandraVectorMemory`.

### Tests
- Added `CassandraVectorMemoryTest` to validate configuration behavior and environment variable fallbacks.
- Added `CassandraVectorMemoryIntegrationTest` (opt-in) that:
  - creates a unique keyspace/table per run,
  - stores/searches vectors end-to-end,
  - performs best-effort teardown cleanup.

### Test gating / CI safety
- Updated Gradle `test` task to **exclude provider tests unless enabled** and real provider credentials exist.
- Updated Gradle `test` task to **exclude vector DB integration tests unless enabled**.
- Updated `BaseIntegrationTest` dotenv loading to ignore missing `.env` to avoid failures when secrets are not present.

### Build / Gradle improvements
- Improved BoxLang JAR resolution logic (handles snapshot vs resolved versions).
- Added explicit Gradle task ordering for test resource downloads/extraction to satisfy Gradle 9 task dependency validation.

### Developer experience
- Added devcontainer support (Cassandra service + postCreate bootstrap script).
- Documented Cassandra schema/index setup and test environment configuration in `readme.md`.

---

### Unit tests (no external services required)
```bash
./gradlew test

## Cassandra integration test (opt-in)

Requires Cassandra with vector support (Cassandra 5.0+ / Astra DB).

```
RUN_VECTOR_DB_TESTS=true \
CASSANDRA_TEST_DATASOURCE=cassandra_vector_test \
CASSANDRA_TEST_CONTACT_POINTS=127.0.0.1:9042 \
./gradlew test --tests "ortus.boxlang.ai.memory.vector.CassandraVectorMemoryIntegrationTest"
```

Type of change

 Improvement
 New Feature
 This change requires a documentation update

Checklist

 My code follows the style guidelines of this project [cfformat](https://github.com/ortus-boxlang/bx-ai/.cfformat.json)
 and [Java](https://github.com/ortus-boxlang/bx-ai/ortus-java-style.xml)
 I have commented my code, particularly in hard-to-understand areas
 I have made corresponding changes to the documentation
 I have added tests that prove my fix is effective or that my feature works
 New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .devcontainer/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+version: "3.9"
```
Comment (lmajano) [inline .devcontainer/docker-compose.yml]: We already have a docker-compose in the root that deals with all dbs, I don't want to introduce yet another way to test them.

File: build.gradle [hunk 2]
```diff
@@ -237,15 +261,89 @@ createModuleStructure.finalizedBy( zipModuleStructure )
...
+
     testLogging {
-		showStandardStreams = true
-        events "passed", "skipped", "failed"
-        exceptionFormat "full"
-        showExceptions true
-        showCauses true
-	}
-	classpath = classpath.filter { !it.path.contains( "build${File.separator}resources" ) }
+        showStandardStreams = true
```
Comment (lmajano) [inline build.gradle]: I really see no value in modifying our build script, when I really want everything to run.  If I want to do targeted tests, we do that via the CLI.

**Review by lmajano — COMMENTED** [review]
1. Leverage the docker compose file we have already for dbs
2. Remove the dev container if not used
3. Everything flows via the dot env usage via .env
4. Restore gralde unless there is a specific reason

**Comment by aykhanpashayev** [comment]
All required changes has been made, devcontainer deleted, docker-compose updated, gradle defaulted

**Comment by lmajano** [comment]
Thanks, I will check this out.
I have been pondering on creating a new `MemoryRegistry` for the next version.  This would allow any third-party memory to be registered, and it would allow us to decouple all these memory implementations from the core and allow the creation of memory modules instead.
